tools/contractdeployer/contract_deployer.go

```diff
@@ -5,37 +5,48 @@ import (
 	"encoding/json"
 	"fmt"
 	"math/big"
+	"strconv"
 	"time"
 
-	"github.com/obscuronet/go-obscuro/integration/guessinggame"
-
+	"github.com/obscuronet/go-obscuro/contracts/managementcontract"
 	"github.com/obscuronet/go-obscuro/integration/erc20contract"
-
-	"github.com/ethereum/go-ethereum/crypto"
-	"github.com/obscuronet/go-obscuro/go/ethadapter/mgmtcontractlib"
+	"github.com/obscuronet/go-obscuro/integration/guessinggame"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/obscuronet/go-obscuro/go/common/log"
 	"github.com/obscuronet/go-obscuro/go/ethadapter"
 	"github.com/obscuronet/go-obscuro/go/wallet"
 )
 
+// The types of contracts supported by the deployer
 const (
 	mgmtContract         = "MGMT"
 	erc20Contract        = "ERC20"
 	guessingGameContract = "GUESS"
-	timeoutWait          = 80 * time.Second
-	retryInterval        = 2 * time.Second
 )
 
-type ContractDeployer struct {
+const (
```

Comment:
@tudor-malene Should be a single const block?

Reply:
I split it in 2, to separate the types of supported contracts (an enum in a normal language), from the other stuff

Reply:
Ah yeah